Tutorial for maps of Mars using pygmt

andrebelem opened the pull request:
There are two versions: the one compatible with the video and the extended version with a 3D comparison with Hawaii and the global Mars map with the Zhurong rover landsite map.

feel free to sort, cut, rearrange as best you can for the course.

andrebelem committed:
Tutorial for maps of Mars using pygmt

There are two versions: the one compatible with the video and the extended version with a 3D comparison with Hawaii and the global Mars map with the Zhurong rover landsite map.

Commit:
Add tutorials to toctree with underscores and script magic

Commit:
Clear cell outputs

maxrjones commented on book/mars_maps.ipynb:
```diff
@@ -0,0 +1,188 @@
+    "\n",
+    "First, we open the `mola32.nc` file using xarray. Note the longitudes are from 0-360°, latitudes are distributed from North to South and the `alt`variable is the MOLA Topography at 32 pixels/degree built from original MOLA file `megt90n000fb.img`. The source is the Mars Climate Database from the European Space Agency."
+   ]
+  },
+  {
```
Suggest including this cell in both notebooks so that they are independent (i.e., data download step is included in both)
```suggestion
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2f913d8d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Also, if you are in colab or trying from your jupyter, you will need the Mars Topography (MOLA) already in Netcdf\n",
    "# a copy of the original file distributed from the Mars Climate Database,\n",
    "# from the European Space Agency under ESTEC contract 11369/95/NL/JG(SC) and Centre National D'Etude Spatial\n",
    "# is in the gdrive.\n",
    "\n",
    "!gdown 1fDzz8AxR1T58y0IGPhmbb1ZwrTLckp2G"
   ]
  },
  {
```

maxrjones commented on book/mars_maps_extended.ipynb:
```diff
@@ -0,0 +1,600 @@
+   },
+   "outputs": [],
+   "source": [
+    "# get etopo around Hawaii \n",
```
```suggestion
    "# get SRTM around Hawaii \n",
```

maxrjones commented on book/mars_maps_extended.ipynb:
```diff
@@ -0,0 +1,600 @@
+   },
+   "outputs": [],
+   "source": [
+    "%%script echo skipping\n",
```
```suggestion
```
I think this cell magic will need to be removed for the jupyter book to deploy properly

maxrjones commented on book/mars_maps.ipynb:
```diff
@@ -0,0 +1,188 @@
+   "id": "e9d6ef48",
+   "metadata": {},
+   "source": [
+    "Just like any other notebook with pygm, we import the library and manipulate other data. To make a map of the entire Martian surface without a lot of time and memory, let's reduce the resolution using `grdsample`. We also take the opportunity to transform an `alt` variable into a `float` to be used in maps."
```
```suggestion
    "Just like any other notebook with pygmt, we import the library and manipulate other data. To make a map of the entire Martian surface without a lot of time and memory, let's reduce the resolution using `grdsample`. We also take the opportunity to transform an `alt` variable into a `float` to be used in maps."
```

maxrjones commented on book/mars_maps_extended.ipynb:
```diff
@@ -0,0 +1,600 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Just like any other notebook with pygm, we import the library and manipulate other data. To make a map of the entire Martian surface without a lot of time and memory, let's reduce the resolution using `grdsample`. We also take the opportunity to transform an `alt` variable into a `float` to be used in maps."
```
```suggestion
    "Just like any other notebook with pygmt, we import the library and manipulate other data. To make a map of the entire Martian surface without a lot of time and memory, let's reduce the resolution using `grdsample`. We also take the opportunity to transform an `alt` variable into a `float` to be used in maps."
```

maxrjones commented on book/mars_maps_extended.ipynb:
```diff
@@ -0,0 +1,600 @@
+    "# convert from int16 to float\n",
+    "dset_mars_topo = dset_mars.alt.astype(float)\n",
+    "\n",
+    "# May be a global Mars map is very insteresting. We just need to get a better resolution not to consume all memory\n",
```
```suggestion
    "# May be a global Mars map is very interesting. We just need to get a better resolution not to consume all memory\n",
```

maxrjones commented on book/mars_maps_extended.ipynb:
```diff
@@ -0,0 +1,600 @@
+    "\n",
+    "**A few notes**\n",
+    "\n",
+    "`zsize` is a bit critical here because the volcano is very big (28 km if we consider -5000 to +23000 m). Likewise, `perspective=[150.45]` was choosen attempting (it's a matter of taste) and depends of which flank of the volcano you want to show. But this choice has to be made according to `shading` since to give a good 3D impression, the lighting must be adjusted according to the elevation a ...
```
```suggestion
    "`zsize` is a bit critical here because the volcano is very big (28 km if we consider -5000 to +23000 m). Likewise, `perspective=[150.45]` was chosen attempting (it's a matter of taste) and depends of which flank of the volcano you want to show. But this choice has to be made according to `shading` since to give a good 3D impression, the lighting must be adjusted according to the elevation and azimuth of the perspective. Finally, the pen outline is made smooth and small to enhance the contours of the topography.\n",
```

maxrjones commented on book/mars_maps.ipynb:
```diff
@@ -0,0 +1,188 @@
+    "\n",
+    "**A few notes**\n",
+    "\n",
+    "`zsize` is a bit critical here because the volcano is very big (28 km if we consider -5000 to +23000 m). Likewise, `perspective=[150,45]` was choosen attempting (it's a matter of taste) and depends of which flank of the volcano you want to show. But this choice has to be made according to `shading` since to give a good 3D impression, the lighting must be adjusted according to the elevation a ...
```
```suggestion
    "`zsize` is a bit critical here because the volcano is very big (28 km if we consider -5000 to +23000 m). Likewise, `perspective=[150,45]` was chosen attempting (it's a matter of taste) and depends of which flank of the volcano you want to show. But this choice has to be made according to `shading` since to give a good 3D impression, the lighting must be adjusted according to the elevation and azimuth of the perspective. Finally, the pen outline is made smooth and small to enhance the contours of the topography.\n",
```

maxrjones reviewed (commented):
Looks good to me to merge after a few typo corrections and enabling the gdown command in both notebooks (included in the review suggestions).

weiji14 committed:
Apply suggestions from code review

Co-authored-by: Max Jones <meghanj@alum.mit.edu>

weiji14 committed:
Fix typo in _toc.yml and reorder sections

weiji14 committed:
Hide stdout of gdown command

Following https://jupyterbook.org/en/stable/content/execute.html#dealing-with-code-that-produces-stdout

weiji14 reviewed (approved):
Cool, thanks Max for the tidying up the notebook. I'll merge this in now and wrap up #10 later today. Can't believe how fast time flies.

@andrebelem, could you let us know the original source of the mola32.nc file? I tried looking online and found https://tharsis.gsfc.nasa.gov/MOLA/mola.php but every dataset link on there is broken. Also saw https://doi.org/10.1126/science.284.5419.1495 but it's behind a paywall and I'm not sure if that's the correct reference.
